drivers: w1: updates to reflect acquistion of maxim by analog devices

str4t0m opened the pull request:
After the acquisition of Maxim Integrated the documentation of these devices has been moved to the analog.com website. Redirects exist, so they are not broken yet, but we should not rely on that forever.
Therefore the links in the drivers and the sensor sample are updated.

MaureenHelm reviewed (dismissed):
Thanks!

kartben commented on drivers/sensor/maxim/ds18b20/ds18b20.c:
```diff
@@ -5,12 +5,10 @@
- *
- * Driver also support the older DS18S20 1-Wire temperature sensors.
+ * Driver for DS18B20 and DS18S20 1-Wire temperature sensors
+ * A datasheets is available at:
```
```suggestion
 * A datasheet is available at:
```

str4t0m replied:
Good catch! As now both datasheets are listed below what do you think about:
` Datasheets for the compatible sensors are available at:`

kartben reviewed (changes requested):
Nit: you may want to split URL updates and other Maxim->Analog renames in different commits, or just reword the single commit message so that it's better aligned with the nature of the update (see e.g. some C files where there's no URL being updated, contrary to what the commit message says)

Thanks for this!

str4t0m committed:
drivers: w1: updates to reflect analog maxim acquisition

Links and the manufacturer name are updated from maxim to analog
for the 1-wire subsystem and the related ds18b20 sensor.

After the acquisition of Maxim Integrated the documentation
of these devices has been moved to the analog.com website.
Redirects exist, so they are not broken yet,
but we should not rely on that.

Signed-off-by: Thomas Stranger <thomas.stranger@outlook.com>

str4t0m committed:
samples: sensor: ds18b20: updates to reflect analog maxim acquisition

Links and the manufacturer name are updated from maxim to analog.

After the acquisition of Maxim Integrated the documentation
of these devices has been moved to the analog.com website.
Redirects exist, so they are not broken yet,
but we should not rely on that.

Signed-off-by: Thomas Stranger <thomas.stranger@outlook.com>

str4t0m commented:
> Nit: you may want to split URL updates and other Maxim->Analog renames in different commits, or just reword the single commit message so that it's better aligned with the nature of the update (see e.g. some C files where there's no URL being updated, contrary to what the commit message says)

I started to just update the links, then noticed a few additional things.

I've decided to just update the commit message and additionally split off the changes in the sample into a separate commit. I hope the description is now better reflecting the changes done.
